project/src/UI/Upgrades/UpgradeButton.gd

```diff
@@ -3,23 +3,84 @@
 tool
 extends TextureButton
 
+signal on_ani_complete(is_entry)
+
 export var texture: Texture setget set_texture
 export var text := "" setget set_text
 
 onready var texture_rect := $VBoxContainer/TextureRect
 onready var label := $VBoxContainer/Label
 
+onready var tween := $Tween
+
+onready var on_pos = self.rect_position
+onready var off_pos = self.rect_position + Vector2(0, 200);
+
+onready var is_entry_ani = true;
+
+
+func animate(entry: bool, delay: float) -> void:
+	self.is_entry_ani = entry;
+	var start_pos = off_pos if entry else on_pos
+	var end_pos = on_pos if entry else off_pos
+	var start_scale = Vector2.ZERO if entry else Vector2.ONE
+	var end_scale = Vector2.ONE if entry else Vector2.ZERO
+	var start_color = Color.transparent if entry else Color.white
+	var end_color = Color.white if entry else Color.transparent
+	var easing = Tween.EASE_OUT if entry else Tween.EASE_IN
+	var fade_time = 0.5 if entry else 0.7
+	self.rect_position = start_pos
+	self.rect_scale = start_scale
+	self.modulate = start_color
+
+	tween.interpolate_property(
```

Comment:
Tweens are not a great tool for animations that involve multiple properties. I would rather recommend using an animation player. Declarative code like the data-based anim player gives you no room for bugs or typo and presents the animation in a more readable fashion.

Doing so would also remove the extra properties you added to the class.

Reply:
Okay, I've made an update and used the animation player instead of using a Tween.